.openshift-ci/sensorversions.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+"""
+Returns the latest patches of the last 4 major versions
+"""
+
+import sys
+import re
+import requests
+import json
+import subprocess
+
+def __isReleaseTagQuay(version):
+    return bool(re.search(r"\d+\.\d+\.\d+$", version))
+
+def __isReleaseTagGitAPI(version):
+    return bool(re.search(r"^refs/tags/\d+\.\d+\.\d+$", version))
+
+def __isReleaseTagGitCLI(version):
+    return bool(re.search(r"^\d+\.\d+\.\d+$", version))
+
+def __filterQuayTags(tags):
+    filteredtags = []
+    for t in tags:
+        name = t['name']
+        if __isReleaseTagQuay(name):
+            filteredtags.append(name)
+    return filteredtags
+
+def __filterGitTags(rawtags):
+    filteredtags = []
+    for t in rawtags:
+        name = t['ref']
+        if __isReleaseTagGitAPI(name):
+            filteredtags.append(name)
+    return set(filteredtags)
+
+def __filterGitCLITags(rawtags):
+    filteredtags = []
+    for t in rawtags:
+        if __isReleaseTagGitCLI(t):
+            filteredtags.append(t)
+    return set(filteredtags)
+
+def __queryQuayForTags():
+    tags = []
+    page = 1
+    pageNotEmpty = True
+    while pageNotEmpty:
+        print("Going through page " + str(page))
+        apiresponse = requests.get("https://quay.io/api/v1/repository/stackrox-io/main/tag/?page=" + str(page) + "&limit=100")
+        rawtags = apiresponse.json()['tags']
+        tags.extend(__filterQuayTags(rawtags))
+        if not bool(rawtags):
+            pageNotEmpty = False
+        page+=1
+    return set(tags)
+
+def __cliOutputToTags(stdoutput):
+    separated = stdoutput.decode(encoding="utf-8").splitlines()
+    return separated
+
+def __transformTagsToNumbers(tags):
+    numTags = []
+    for t in tags:
+        numTags.append(__splitVersion(t))
+    return sorted(numTags)
+
+def __splitVersion(version):
+    digits = re.search(r"(\d+)\.(\d+)\.\D*(\d+)", version)
+    return int(digits.group(1)), int(digits.group(2)), int(digits.group(3))
+
+def __getLatestTags(current_version, tags, num_versions):
```

Comment:
[nit] I think, leading underscores only clutter the code. In a simple CLI that's not object-oriented, I _think_, it should be fine to write functions as "public", i.e. without leading `__`. No strong opinion though.

Reply:
I learned to do it this way as good form since those functions are not supposed to be called from the outside, I'll change it if that's our common practice though.